### PR TITLE
Open file dialog io exception Fixes #7998

### LINKED ISSUE #7998: Document that OpenFileDialog.OpenFile() throws IOException
`OpenFileDialog.OpenFile()` is a thin wrapper around `new FileStream()`. Thus, it will throw the same exceptions. Unlike `SaveFileDialog.OpenFile()` which behaves differently on mono, this behavior is the same on [.netfx](https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/WinForms/OpenFileDialog.cs,b9fb6d8411201152), [.netcore](https://github.com/dotnet/winforms/blob/6a3867101ca058010147742789d7ab52b11da0bd/src/System.Windows.Forms/src/System/Windows/Forms/OpenFileDialog.cs#L100), and [mono](https://github.com/mono/mono/blob/main/mcs/class/System.Windows.Forms/System.Windows.Forms/OpenFileDialog.cs#L120).

Please document that `OpenFileDialog.OpenFile()` will throw `IOException` if the file cannot be opened, such as if it was removed or renamed prior to calling `OpenFile()`.

### PR DESCRIPTION
**Pull request by BlueBles**
## Summary

Adding information about the exception that may be thrown if the file is modified/deleted before openFile()

Remarks: It is my first PR to this repo.

Fixes #7998

### CODE REVIEW

File: xml/System.Windows.Forms/OpenFileDialog.xml [hunk 1]
```diff
@@ -47,8 +47,8 @@
 On a right-to-left operating system, setting the containing form's <xref:System.Windows.Forms.Control.RightToLeft> property to <xref:System.Windows.Forms.RightToLeft.Yes?displayProperty=nameWithType> localizes the dialog's **File Name**, **Open**, and **Cancel** buttons. If the property is not set to <xref:System.Windows.Forms.RightToLeft.Yes?displayProperty=nameWithType>, English text is used instead.  
 
  If you want to give the user the ability to select a folder instead of a file, use <xref:System.Windows.Forms.FolderBrowserDialog> instead.  
-  
-
```
Comment (Jozkee) [inline xml/System.Windows.Forms/OpenFileDialog.xml:51]: This should be documented as an `<exception>` in the member, not in the type.
https://github.com/dotnet/dotnet-api-docs/blob/5b0810324b50815313c14a862ab4acc2290c9243/xml/System.Windows.Forms/OpenFileDialog.xml#L237

For reference, for [System.IO.File.Open](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.open?view=net-8.0) we say "An I/O error occurred while opening the file" for `IOException`s.

Comment (BlueBles) [reply xml/System.Windows.Forms/OpenFileDialog.xml:51]: Thank you for information. I did as you suggested :)

**Comment by BlueBles** [comment]
@Jozkee could you review?

**Review by Jozkee — APPROVED** [review]
Thanks.

**Pull request merged**